packages/blade/src/components/Badge/Badge.tsx

```diff
@@ -46,6 +64,40 @@ type BadgeProps = {
   fontWeight?: 'regular' | 'bold';
 };
 
+type ColorProps = {
+  iconColor: IconProps['color'];
+  textColor: BaseTextProps['color'];
+  backgroundColor: FeedbackBackgroundColors | BadgeBackgroundColors;
+  borderColor: FeedbackBorderColors | BadgeBorderColors;
+};
+const getColorProps = ({
+  variant,
+  contrast,
+}: {
+  variant: NonNullable<BadgeProps['variant']>;
+  contrast: NonNullable<BadgeProps['contrast']>;
+}): ColorProps => {
+  // const feedbackVariants = ['information', 'negative', 'neutral', 'notice', 'positive'] as const;
+  const props: ColorProps = {
+    iconColor: 'feedback.icon.neutral.lowContrast',
+    textColor: 'feedback.text.neutral.lowContrast',
+    backgroundColor: 'feedback.background.neutral.lowContrast',
+    borderColor: 'feedback.border.neutral.lowContrast',
+  };
+  if (variant != 'blue') {
```

Comment:
@anuraghazra know any better way to handle this? I ideally wanted to do 
```jsx
  const feedbackVariants = ['information', 'negative', 'neutral', 'notice', 'positive']

  if (feedbackVariants.includes(variant)) {
    props.iconColor = `feedback.icon.${variant}.${contrast}Contrast`;
     ...
  }
```
But when I do this `variant` doesn't strip away the value `blue` from it which leads to a TS error on `props.iconColor` since it doesn't recognize a valid value with `blue` as variant. When I do `if (variant != 'blue')`, it strips away the type properly.

Reply:
If you don't want to do variant !== 'blue'

you can add a type guard checking for feedbackVariants

```ts
const isFeedbackVariant = (variant: string): variant is Feedback => {
  const feedbackVariants = ['information', 'negative', 'neutral', 'notice', 'positive'] as const;
  return feedbackVariants.includes(variant);
};


if (isFeedbackVariant(variant)) {
  props.iconColor = `feedback.icon.${variant}.${contrast}Contrast`;
  props.textColor = `feedback.text.${variant}.${contrast}Contrast`;
  props.backgroundColor = `feedback.background.${variant}.${contrast}Contrast`;
  props.borderColor = `feedback.border.${variant}.${contrast}Contrast`;
} else {
  props.iconColor = `badge.icon.${variant}.${contrast}Contrast`;
  props.textColor = `badge.text.${variant}.${contrast}Contrast`;
  props.backgroundColor = `badge.background.${variant}.${contrast}Contrast`;
  props.borderColor = `badge.border.${variant}.${contrast}Contrast`;
}
```

Reply:
Nice